File: src/ios/BarcodeType.swift
//
//  BarcodeType.swift
//  MLBarcodeScanner
//
//  Created by Felix Nievelstein on 15.11.21.
//

import Foundation
import MLKit

@objc enum BarcodeType: Int, Codable {
    case QR_CODE
    case EAN_13
    case EAN_8
    case Unknown
    
    static func from(format: BarcodeFormat) -> BarcodeType {
        switch format {
        case .EAN13:
                return EAN_13
        case .EAN8:
                return EAN_8
        case .qrCode:
                return QR_CODE
        default:
            return Unknown
        }
    }
}

File: src/android/BarcodeType.kt
package de.impacgroup.mlbarcodescanner.module

import com.google.mlkit.vision.barcode.Barcode

enum class BarcodeType {
    QR_CODE,
    EAN_13,
    EAN_8,
    Unknown;

    companion object {
        fun from(barcode: Barcode): BarcodeType {
            return when(barcode.format) {
                Barcode.FORMAT_EAN_13 -> EAN_13
                Barcode.FORMAT_EAN_8 -> EAN_8
                Barcode.FORMAT_QR_CODE -> QR_CODE
                else -> Unknown
            }
        }
    }

}